Batch 1 of RN BZ doc text (#48262)
openshift/openshift-docs: release_notes/ocp-4-11-release-notes.adoc

[id="ocp-4-11-olm-bug-fixes"]
==== Operator Lifecycle Manager (OLM)

> * Before this update, invalid subscription labels were created when a resource name exceeded 63 characters. Truncating labels that exceed the 63 character limit resolves the issue, and the subscription resource no longer rejects the Kubernetes API. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2016425[*BZ#2016425*])

Reviewer: ```suggestion
* Before this update, invalid subscription labels were created when a resource name exceeded 63 characters. Truncating labels that exceed the 63-character limit resolves the issue, and the subscription resource no longer rejects the Kubernetes API. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2016425[*BZ#2016425*])
```

Reviewer: ISG: https://www.ibm.com/docs/en/ibm-style?topic=punctuation-hyphens#compound-words

* Before this update, catalog source pods for the Marketplace Operator prevented nodes from draining. As a result, the Cluster Autoscaler could not scale down effectively. With this update, adding the `cluster-autoscaler.kubernetes.io/safe-to-evict` annotation to the catalog source pods fixes the issue, and the Cluster Autoscaler can scale down effectively.(link:https://bugzilla.redhat.com/show_bug.cgi?id=2053343[*BZ#2053343*])

* Before this update, the `collect-profiles` job could take a long time to complete in certain circumstances, such as when a pod could not be scheduled. As a result, if enough jobs were scheduled but unable to run, the number of scheduled jobs exceeded pod quota limits. With this update, only one `collect-profiles` pod exists at a time, and the `collect-profiles` job does not exceed pod quota limits. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2055861[*BZ#2055861*])

> * Before this update, the package server was not aware of pod topology when defining its leader election duration, renewal deadline, and retry periods. As a result, the package server strained topologies with limited resources, such as single node environments. This update introduces a `leaderElection` package that sets reasonable lease duration, renewal deadlines, and retry periods. This fix reduces strain on clusters with limited resources. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2048563[*BZ#2048563*])

Reviewer 2: ```suggestion
* Before this update, the package server was not aware of pod topology when defining its leader election duration, renewal deadline, and retry periods. As a result, the package server strained topologies with limited resources, such as single-node environments. This update introduces a `leaderElection` package that sets reasonable lease duration, renewal deadlines, and retry periods. This fix reduces strain on clusters with limited resources. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2048563[*BZ#2048563*])
```

> * Previously, there was a bad catalog source in the openshift-marketplace. Because of this, all subscriptions were blocked. With this update, if there is a bad catalog source in the openshift-marketplace, users can subscribe to an operator from a quality catalog source of their own namespace with the OG annotation. As a result, if there is a bad catalog source in the local namespace, the user cannot subscribe to any operator in the namespace. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2076323[*BZ2076323*]

Author: Missed the closed parenthesis. Will catch after the review.
```suggestion
* Previously, there was a bad catalog source in the openshift-marketplace. Because of this, all subscriptions were blocked. With this update, if there is a bad catalog source in the openshift-marketplace, users can subscribe to an operator from a quality catalog source of their own namespace with the OG annotation. As a result, if there is a bad catalog source in the local namespace, the user cannot subscribe to any operator in the namespace. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2076323[*BZ2076323*])
```

Reviewer 2: This looks like a namespace, based on the bug. Also, I'm pretty sure they mean original for OG.

```suggestion
* Previously, there was a bad catalog source in the `openshift-marketplace` namespace. Because of this, all subscriptions were blocked. With this update, if there is a bad catalog source in the `openshift-marketplace` namespace, users can subscribe to an operator from a quality catalog source of their own namespace with the original annotation. As a result, if there is a bad catalog source in the local namespace, the user cannot subscribe to any operator in the namespace. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2076323[*BZ2076323*])
```

> * Previously, info level logs were being generated during operator-marketplace polling, which caused log spam. This update uses the command line flag to reduce the log line to the debug level and adds more control of the log levels for the user. As a result, this reduces log spam. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2057558[*BZ#2057558*]

Author: missed closed parenthesis here too.

> * Previously, info level logs were being generated during operator-marketplace polling, which caused log spam. This update uses the command line flag to reduce the log line to the debug level and adds more control of the log levels for the user. As a result, this reduces log spam. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2057558[*BZ#2057558*]

Reviewer: ```suggestion
* Previously, info-level logs were being generated during operator-marketplace polling, which caused log spam. This update uses the command line flag to reduce the log line to the debug level, and adds more control of the log levels for the user. As a result, this reduces log spam. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2057558[*BZ#2057558*]
```

Reviewer: ISG: https://www.ibm.com/docs/en/ibm-style?topic=punctuation-commas#commas-between-clauses

> * Previously, info level logs were being generated during operator-marketplace polling, which caused log spam. This update uses the command line flag to reduce the log line to the debug level and adds more control of the log levels for the user. As a result, this reduces log spam. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2057558[*BZ#2057558*]

Reviewer: ```suggestion
* Previously, info-level logs were being generated during operator-marketplace polling, which caused log spam. This update uses the command line flag to reduce the log line to the debug level and adds more control of the log levels for the user. As a result, this reduces log spam. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2057558[*BZ#2057558*]
```

> * Previously, info level logs were being generated during operator-marketplace polling, which caused log spam. This update uses the command line flag to reduce the log line to the debug level and adds more control of the log levels for the user. As a result, this reduces log spam. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2057558[*BZ#2057558*]

Reviewer 2: ```suggestion
* Previously, info-level logs were generated during `operator-marketplace` project polling, which caused log spam. This update uses the command line flag to reduce the log line to the debug level and adds more control of the log levels for the user. As a result, this reduces log spam. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2057558[*BZ#2057558*]
```

> * Previously, each component managed by CVO consists of yamls defined in the `/manifest` directory in the root of a project's repo. When removing a yaml from the `/manifest` directory, you had to add the `release.openshift.io/delete: “true”` annotation, otherwise the CVO would not delete the resources from the cluster. This update reintroduces any resources that were removed from the `/manifest` and includes the `release.openshift.io/delete: “true”` annotation so that the CVO cleans up the resources. As a result, resources that are no longer required for the OLM component are removed from the customers' clusters.(link:https://bugzilla.redhat.com/show_bug.cgi?id=1975543[*BZ#1975543*])

Reviewer: ```suggestion
* Previously, each component managed by CVO consists of yaml files defined in the `/manifest` directory in the root of a project's repo. When removing a yaml file from the `/manifest` directory, you had to add the `release.openshift.io/delete: “true”` annotation, otherwise the CVO would not delete the resources from the cluster. This update reintroduces any resources that were removed from the `/manifest` and includes the `release.openshift.io/delete: “true”` annotation so that the CVO cleans up the resources. As a result, resources that are no longer required for the OLM component are removed from the customers' clusters.(link:https://bugzilla.redhat.com/show_bug.cgi?id=1975543[*BZ#1975543*])
```

> * Previously, the CheckRegistryServer function used by GRPC catalogSources did not confirm the existence of the serviceAccount associated with the catalogSource. This caused the existence of an unhealthy catalogSource with no serviceAccount. With this update, the GRPC CheckRegistryServer function checks if the serviceAccount exists, which will recreate the service if it's not found. As a result, the OLM will recreate serviceAccounts owned by GRPC catalogSouces if they do not exist. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1975543[*BZ#1975543*])

Reviewer: ```suggestion
* Previously, the CheckRegistryServer function used by GRPC catalogSources did not confirm the existence of the serviceAccount associated with the catalogSource. This caused the existence of an unhealthy catalogSource with no serviceAccount. With this update, the GRPC CheckRegistryServer function checks if the serviceAccount exists, which will recreate the service if it is not found. As a result, the OLM will recreate serviceAccounts owned by GRPC catalogSouces if they do not exist. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1975543[*BZ#1975543*])
```

Reviewer 2: ```suggestion
* Previously, the CheckRegistryServer function used by GRPC catalog sources did not confirm the existence of the service account associated with the catalog source. This caused the existence of an unhealthy catalog source with no service account. With this update, the GRPC `CheckRegistryServer` function checks if the service account exists and recreates the service if it is not found. As a result, the OLM recreats service accounts owned by GRPC catalog souces if they do not exist. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1975543[*BZ#1975543*])
```

Please review the [API object reference](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#api-object-formatting) in the contrib guide.

[id="ocp-4-11-web-console-developer-perspective-bug-fixes"]
==== Web console (Developer perspective)

> * Before this update, the *Git Import* form displayed an error message when an invalid devfile repository (older than devfile v2.2) is entered. With this update, the error message state that devfiles older than v2.2 is not supported. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2046435[*BZ#2046435*])

Reviewer: ```suggestion
* Before this update, the *Git Import* form displayed an error message when an invalid devfile repository (older than devfile v2.2) is entered. With this update, the error message states that devfiles older than v2.2 are not supported. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2046435[*BZ#2046435*])
```

> * Before this update, if the *ConsoleLink CR* (openshift-blog) was not available in the cluster that causes the blog link to be undefined. So while clicking on the blog link did not redirect to the openshift blog. With this update fall back link to https://developers.redhat.com/products/openshift/whats-new is added even if the *ConsoleLink CR* (openshift-blog) is not present in the cluster. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2050637[BZ#2050637])

Reviewer: ```suggestion
* Before this update, the *ConsoleLink CR* (openshift-blog) was not available in the cluster that causing the blog link to be undefined. So while clicking the blog link, it did not redirect to the openshift blog. With this update, the fall back link to https://developers.redhat.com/products/openshift/whats-new is added even if the *ConsoleLink CR* (openshift-blog) is not present in the cluster. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2050637[BZ#2050637])
```

Reviewer 2: ```suggestion
* Before this update, if the *ConsoleLink CR* (openshift-blog) was not available in the cluster, the blog link was undefined. Clicking the blog link did not redirect to the OpenShift blog. With this update, a fall back link to https://developers.redhat.com/products/openshift/whats-new is added even if the *ConsoleLink CR* (openshift-blog) is not present in the cluster. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2050637[BZ#2050637])
```

> * Before this update, if the *ConsoleLink CR* (openshift-blog) was not available in the cluster that causes the blog link to be undefined. So while clicking on the blog link did not redirect to the openshift blog. With this update fall back link to https://developers.redhat.com/products/openshift/whats-new is added even if the *ConsoleLink CR* (openshift-blog) is not present in the cluster. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2050637[BZ#2050637])

Reviewer 3: iiuc, consider the following rewording:
```suggestion
* Before this update, if the *ConsoleLink CR* (openshift-blog) was not available in the cluster, it would cause the blog link to be undefined. This would prevent a user from being redirected to the {product-title} blog if they clicked on the blog link. With this update, a fall back link to https://developers.redhat.com/products/openshift/whats-new is added even if the *ConsoleLink CR* (openshift-blog) is not present in the cluster. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2050637[BZ#2050637])
```

> * Before this update, the API version for kafka CR got updated that did not support the old version which used to show empty *Bootstrap server* dropdown in *Create Event Source - KafkaSource* form even if it was created. With this update, the updated APIs for Kafka CR supports the old versions and renders the *Bootstrap server* dropdown in *Create Event Source - KafkaSource* form.(link:https://bugzilla.redhat.com/show_bug.cgi?id=2058623[BZ#2058623])

Reviewer: ```suggestion
* Before this update, the API version for kafka CR got updated that did not support the old version, which used to show empty *Bootstrap server* dropdown in *Create Event Source - KafkaSource* form even if it was created. With this update, the updated APIs for Kafka CR supports the old versions, and renders the *Bootstrap server* dropdown in the *Create Event Source - KafkaSource* form.(link:https://bugzilla.redhat.com/show_bug.cgi?id=2058623[BZ#2058623])
```

Reviewer 2: Maybe
```suggestion
* Before this update, the API version for the Kafka CR was updated. This version did not support the old version, so an empty *Bootstrap server* was displayed *Create Event Source - KafkaSource* form even if it was created. With this update, the updated APIs for the Kafka CR support the old versions and renders the *Bootstrap server* list in the *Create Event Source - KafkaSource* form. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2058623[BZ#2058623])
```

> * Before this update, when you use the *Import from Git* form to import a private git repository, the correct import type and a builder image were not identified. Because the secret to fetch the private repository details is not decoded. With this update the *Import from Git* form decodes the secret to fetch the private repository details. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2053501[BZ#2053501])

Reviewer: Should the sentence starting with 'Because the secret...' be a part of one of the sentences around it?

> * Before this update, when you use the *Import from Git* form to import a private git repository, the correct import type and a builder image were not identified. Because the secret to fetch the private repository details is not decoded. With this update the *Import from Git* form decodes the secret to fetch the private repository details. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2053501[BZ#2053501])

Reviewer: ```suggestion
* Before this update, when you use the *Import from Git* form to import a private git repository, the correct import type and a builder image were not identified. Because the secret to fetch the private repository details is not decoded. With this update, the *Import from Git* form decodes the secret to fetch the private repository details. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2053501[BZ#2053501])
```

> * Before this update, when you use the *Import from Git* form to import a private git repository, the correct import type and a builder image were not identified. Because the secret to fetch the private repository details is not decoded. With this update the *Import from Git* form decodes the secret to fetch the private repository details. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2053501[BZ#2053501])

Reviewer 2: ```suggestion
* Before this update, when you used the *Import from Git* form to import a private Git repository, the correct import type and a builder image were not identified because the secret to fetch the private repository details was not decoded. With this update the *Import from Git* form decodes the secret to fetch the private repository details. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2053501[BZ#2053501])
```

> * Before this update, when you use the *Import from Git* form to import a private git repository, the correct import type and a builder image were not identified. Because the secret to fetch the private repository details is not decoded. With this update the *Import from Git* form decodes the secret to fetch the private repository details. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2053501[BZ#2053501])

Reviewer 3: ```suggestion
* Before this update, when you use the *Import from Git* form to import a private git repository, the correct import type and a builder image could not be identified because the secret to fetch the private repository details was not decoded. With this update the *Import from Git* form decodes the secret to fetch the private repository details. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2053501[BZ#2053501])
```

> * Before this update, from the developer perspective, the *Observe* dashboard opens for the most recently viewed workload rather than the one you selected in the *Topology*. This issue happens because the session prefers the redux store instead of the query parameters in the URL. With this update, *Observe* dashboard renders components baked on the query parameters in the URL. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2052953[BZ#2052953])

Reviewer 2: ```suggestion
* Before this update, from the developer perspective, the *Observe* dashboard opened for the most recently viewed workload rather than the one you selected in the *Topology* view. This issue happens because the session prefers the redux store instead of the query parameters in the URL. With this update, the *Observe* dashboard renders components based on the query parameters in the URL. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2052953[BZ#2052953])
```

> * Before this update, the *Pipeline* used to start with the hardcoded value `gp2` as the default storage class even if it did not exist on the cluster. With this update you can use the default specified storage class name instead of a hardcoded value. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2084635[BZ#2084635])

Reviewer: ```suggestion
* Before this update, the *Pipeline* used to start with the hardcoded value `gp2` as the default storage class even if it did not exist on the cluster. With this update, you can use the default specified storage class name instead of a hardcoded value. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2084635[BZ#2084635])
```
